Source/objects.cpp

```diff
@@ -356,55 +356,39 @@ int StoryText[3][3] = {
 
 void __cdecl InitObjectGFX()
 {
-	ObjDataStruct *v0;          // eax
-	char *v1;                   // esi
-	unsigned char v2;           // cl
-	int v3;                     // edx
-	int i;                      // eax
-	char v5;                    // al
-	signed int v7;              // ebx
-	char *v8;                   // ST08_4
-	unsigned char *v9;          // eax
-	int v10;                    // ecx
-	unsigned char fileload[56]; // [esp+4h] [ebp-58h]
-	char filestr[32];           // [esp+3Ch] [ebp-20h]
-
-	memset(fileload, 0, sizeof(fileload));
-	if (AllObjects[0].oload != -1) {
-		v0 = AllObjects;
-		v1 = &AllObjects[0].otheme;
-		do {
-			if (v0->oload == 1 && currlevel >= (signed int)(char)*(v1 - 3) && currlevel <= (signed int)(char)*(v1 - 2))
-				fileload[(char)*(v1 - 4)] = 1;
-			v2 = *v1;
-			if (*v1 != -1) {
-				v3 = numthemes;
-				for (i = 0; i < v3; ++i) {
-					if (themes[i].ttype == v2)
-						fileload[(char)*(v1 - 4)] = 1;
-				}
-			}
-			v5 = v1[1];
-			if (v5 != -1) {
-				if (QuestStatus(v5))
-					fileload[(char)*(v1 - 4)] = 1;
+	BOOLEAN fileload[56];
+	char filestr[32];
+	int i, j;
+
+	memset(fileload, FALSE, sizeof(fileload));
+
+	for (i = 0; AllObjects[i].oload != -1; i++) {
+		if (AllObjects[i].oload == 1
+		    && (signed int)currlevel >= AllObjects[i].ominlvl

```

Comment:
```suggestion
		    && (int)currlevel >= AllObjects[i].ominlvl
```
Or even better, if possible:
```suggestion
		    && currlevel >= AllObjects[i].ominlvl
```

Reply:
if I left a conversion, it means it's necessary

Reply:
Ok, I some times miss stuff like that :)